src/index.ts

```diff
@@ -138,6 +140,66 @@ export const createInstance = (config: Config): Instance => {
 
       return tx;
     },
+
+    buyNow: async (
+      signer: ethers.Signer,
+      params: BuyNowParams
+    ): Promise<ethers.ContractTransaction> => {
+      const zAuction = await getZAuctionContract(
+        signer,
+        config.zAuctionAddress
+      );
+
+      const nftContract = await getERC721Contract(
+        config.web3Provider,
+        config.tokenContract
+      );
+
+      const isApproved = await actions.isZAuctionApprovedNftTransfer(
+        await signer.getAddress(),
```

Comment:
This is the wrong address to be checking this on.

The `signer` in this case is the `buyer`. The buyer doesn't need to approve zAuction to transfer NFT's, the seller does.
Instead you should be getting the `seller` in this context (current owner of the domain via `nftContract.ownerOf(params.tokenId)` and making sure they approved zAuction to transfer NFT's.

If they didn't, the error message should be `Seller did not approve zAuction to transfer NFT`